classlib/src/main/java/org/teavm/classlib/java/lang/TFloat.java

```diff
@@ -84,7 +85,7 @@ public boolean equals(Object other) {
         if (this == other) {
             return true;
         }
-        return other instanceof TFloat && ((TFloat) other).value == value;
+        return other instanceof TFloat && floatToIntBits(((TFloat) other).value) == floatToIntBits(value);
```

Comment:
I'm not sure if it has good approach WRT performance. Also, I'd check various cases with infinity and NaN.

Reply:
This is how equals works in JDK. It compares doubles bitwise, not with == operator.
So, Float.NaN != Float.NaN, but Float.valueOf(Float.NaN).equals(Float.NaN).
I don't know why this is done that way, but I think that main reason for this is to make equality relation reflexive and therefore equivalence relation (which would not be in case of comparing by == for NaN).
I'm not insisting on this change, I never do Map<Double, Something>, but this is how it works in JDK.
Added test cases which cover this behavior (will be failing when rollback this change).

Reply:
By the way, hashCode is done same way: floatToIntBits(value).

Reply:
One more point to follow consistency.